pennylane/_qubit_device.py

```diff
@@ -1437,6 +1437,8 @@ def circuit(x):
 
         if isinstance(obs, CountsMP):
             # convert samples and outcomes (if using) from arrays to str for dict keys
+            samples = np.array([sample for sample in samples if not np.any(np.isnan(sample))])
```

Comment:
Should we also have this fix in `CountsMP.process_samples`?

Reply:
I think it's a good idea, I'll add that and a couple of tests